lib/Github/Api/GitData/References.php

```diff
@@ -25,6 +25,34 @@ public function all($username, $repository)
         return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/refs');
     }
 
+    /**
+     * Get all matching references for a particular branch of a repository.
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $branch
+     *
+     * @return array
+     */
+    public function matchingBranch($username, $repository, $branch)
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/matching-refs/heads/'.$branch);
+    }
+
+    /**
+     * Get all matching references for a particular branch of a repository.
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $tag
+     *
+     * @return array
+     */
+    public function matchingTag($username, $repository, $tag)
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/matching-refs/tags/'.$tag);
```

Comment:
```suggestion
        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/matching-refs/tags/'.rawurlencode($tag));
```

To be sure add a `rawurlencode` around the branch parameter